File: Perl/conintl/ConIntl.pl
use Mojolicious::Lite -signatures;
use POSIX qw(floor);

my @unidades   = ("", "uno", "dos", "tres", "cuatro", "cinco", "seis", "siete", "ocho", "nueve");
my @decenas    = ("", "diez", "veinte", "treinta", "cuarenta", "cincuenta", "sesenta", "setenta", "ochenta", "noventa");
my @especiales = ("diez", "once", "doce", "trece", "catorce", "quince", "dieciséis", "diecisiete", "dieciocho", "diecinueve");
my @centenas   = ("", "cien", "doscientos", "trescientos", "cuatrocientos", "quinientos", "seiscientos", "setecientos", "ochocientos", "novecientos");

sub convertir_bloque ($n) {
    my $res = "";
    if ($n >= 100) {
        return "cien " if $n == 100;
        if ($n > 100 && $n < 200) {
            $res .= "ciento ";
        } else {
            $res .= $centenas[floor($n / 100)] . " ";
        }
        $n %= 100;
    }
    if ($n >= 10 && $n < 20) {
        $res .= $especiales[$n - 10] . " ";
        return $res;
    } elsif ($n >= 20) {
        return $res . "veinte " if $n == 20;
        return $res . "veinti" . $unidades[$n % 10] . " " if ($n > 20 && $n < 30);
        
        $res .= $decenas[floor($n / 10)];
        $res .= " y " . $unidades[$n % 10] if ($n % 10 != 0);
        $res .= " ";
        return $res;
    }
    if ($n > 0 && $n < 10) {
        $res .= $unidades[$n] . " ";
    }
    return $res;
}

sub numero_a_letras ($n) {
    return "cero" if $n == 0;
    my $resultado = "";

    if ($n >= 1000000) {
        my $millones = floor($n / 1000000);
        if ($millones == 1) {
            $resultado .= "un millón ";
        } else {
            $resultado .= convertir_bloque($millones) . "millones ";
        }
        $n %= 1000000;
    }
    if ($n >= 1000) {
        my $miles = floor($n / 1000);
        if ($miles == 1) {
            $resultado .= "mil ";
        } else {
            $resultado .= convertir_bloque($miles) . "mil ";
        }
        $n %= 1000;
    }
    $resultado .= convertir_bloque($n);
    
    # Trim
    $resultado =~ s/^\s+|\s+$//g;
    return $resultado;
}

get '/' => sub ($c) {
    my $numero_str = $c->param('n') || "10";
    
    # Validar que sea un número
    if ($numero_str =~ /^\d+$/) {
        my $letras = numero_a_letras($numero_str);
        $c->render(text => $letras, format => 'txt');
    } else {
        $c->render(text => "Por favor, proporcione un número entero válido.", format => 'txt');
    }
};

app->start('daemon', '-l', 'http://*:8000');

File: Rust/soap_rust/src/conIntl.rs
use axum::extract::Query;
use std::collections::HashMap;

pub async fn handler(Query(params): Query<HashMap<String, String>>) -> String {
    let n: u64 = params.get("n")
        .and_then(|s| s.parse().ok())
        .unwrap_or(10);

    format!("{}", convertir_a_letras(n))
}

fn convertir_bloque(n: u64) -> String {
    let unidades = ["", "uno", "dos", "tres", "cuatro", "cinco", "seis", "siete", "ocho", "nueve"];
    let decenas = ["", "diez", "veinte", "treinta", "cuarenta", "cincuenta", "sesenta", "setenta", "ochenta", "noventa"];
    let especiales = ["diez", "once", "doce", "trece", "catorce", "quince", "dieciséis", "diecisiete", "dieciocho", "diecinueve"];
    let centenas = ["", "cien", "doscientos", "trescientos", "cuatrocientos", "quinientos", "seiscientos", "setecientos", "ochocientos", "novecientos"];

    let mut res = String::new();
    let mut num = n;

    if num >= 100 {
        if num == 100 { return "cien ".to_string(); }
        if num > 100 && num < 200 { res.push_str("ciento "); }
        else { res.push_str(centenas[(num / 100) as usize]); res.push(' '); }
        num %= 100;
    }

    if num >= 10 && num < 20 {
        res.push_str(especiales[(num - 10) as usize]);
        res.push(' ');
    } else if num >= 20 {
        if num == 20 { res.push_str("veinte "); }
        else if num > 20 && num < 30 { res.push_str(&format!("veinti{} ", unidades[(num % 10) as usize])); }
        else {
            res.push_str(decenas[(num / 10) as usize]);
            if num % 10 != 0 { res.push_str(&format!(" y {}", unidades[(num % 10) as usize])); }
            res.push(' ');
        }
    } else if num > 0 {
        res.push_str(unidades[num as usize]);
        res.push(' ');
    }
    res
}

fn convertir_a_letras(n: u64) -> String {
    if n == 0 { return "cero".to_string(); }
    let mut resultado = String::new();
    let mut num = n;

    if num >= 1_000_000 {
        let millones = num / 1_000_000;
        resultado.push_str(if millones == 1 { "un millón ".to_string() } else { format!("{}millones ", convertir_bloque(millones)) }.as_str());
        num %= 1_000_000;
    }

    if num >= 1_000 {
        let miles = num / 1_000;
        resultado.push_str(if miles == 1 { "mil ".to_string() } else { format!("{}mil ", convertir_bloque(miles)) }.as_str());
        num %= 1_000;
    }

    resultado.push_str(&convertir_bloque(num));
    resultado.trim().to_string()
}